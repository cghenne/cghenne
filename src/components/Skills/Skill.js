// @flow
import React from 'react';
import './Skills.css';

type Props = {
  label: string,
  list: Array<string>
};

const Skill = ({ label, list }: Props) => {
  return (
    <div className="skill">
      <h4>{label}</h4>
      <ul>
        {list.map((item: string, index: number) => <li key={index}>{item}</li>)}
      </ul>
    </div>
  );
};

Skill.displayName = 'Skill';

export default Skill;
