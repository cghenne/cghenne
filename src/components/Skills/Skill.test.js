import React from 'react';
import { shallow } from 'enzyme';
import Skill from './Skill';

const fakeSkill = {
  label: 'Cooking',
  list: ['Pastry', 'Oven', 'Slow Cooker']
};

describe('Skill', () => {
  const wrapper = shallow(<Skill {...fakeSkill} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the skill in a skill class', () => {
    expect(wrapper.find('div').prop('className')).toEqual('skill');
  });

  it('should render the skill label in an h4 tag', () => {
    expect(wrapper.find('h4').text()).toEqual(fakeSkill.label);
  });

  it('should render a list', () => {
    expect(wrapper.find('ul').exists()).toEqual(true);
  });

  it('should render the list of skill', () => {
    const list = wrapper.find('li');
    expect(list.length).toEqual(fakeSkill.list.length);
    list.map((elem, index) =>
      expect(elem.text()).toEqual(fakeSkill.list[index])
    );
  });
});
