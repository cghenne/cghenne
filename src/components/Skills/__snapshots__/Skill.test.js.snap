// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill should render without crashing 1`] = `
<div
  className="skill"
>
  <h4>
    Cooking
  </h4>
  <ul>
    <li
      key="0"
    >
      Pastry
    </li>
    <li
      key="1"
    >
      Oven
    </li>
    <li
      key="2"
    >
      Slow Cooker
    </li>
  </ul>
</div>
`;
